### ISSUE TITLE
Clicking on a focusable element doesn't focus it

### ISSUE THREAD

**Issue by jdm** [servo/servo#26899]
```
<style>div:focus { color: red; }</style>
<div tabindex=0 style="background-color: green; width: 100px; height: 100px;"><p>hello</p></div>
<button onclick="document.querySelector('div').focus()">focus</button>
```
The `tabindex=0` makes the element focusable. In Firefox clicking on the element turns the text red; in Servo it does not.

**Comment by jdm** [comment]
Clicking on the rest of the page does make the div unfocused if it is focused, however.

**Comment by jdm** [comment]
Fixed by #26902.
